vault/logical_system_namespaces_test.go

```diff
@@ -16,25 +16,35 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func testCreateNamespace(t *testing.T, b logical.Backend, name string, customMetadata map[string]string) {
+func testCreateNamespace(t *testing.T, ctx context.Context, b logical.Backend, name string, customMetadata map[string]string) *namespace.Namespace {
 	t.Helper()
 	req := logical.TestRequest(t, logical.UpdateOperation, path.Join("namespaces", name))
 	if customMetadata != nil {
 		req.Data["custom_metadata"] = customMetadata
 	}
-	_, err := b.HandleRequest(context.Background(), req)
+	if ctx == nil {
+		ctx = namespace.RootContext(context.Background())
+	}
+	res, err := b.HandleRequest(ctx, req)
 	require.NoError(t, err)
+
+	return &namespace.Namespace{
+		ID:             res.Data["id"].(string),
+		Path:           res.Data["path"].(string),
+		CustomMetadata: res.Data["custom_metadata"].(map[string]string),
+	}
 }
 
 func TestNamespaceBackend_Set(t *testing.T) {
 	t.Parallel()
 	b := testSystemBackend(t)
+	rootCtx := namespace.RootContext(context.Background())
```

Comment:
Do we not infer the root context? It seems like this might be a breaking change for testing; we don't export `namespace`, which makes it a bit hard for third-party packages to do testing I think?

I wonder if we can update it to return root namespace if there was no namespace on the request?

Reply:
There was no inferring going on. The implementation just assumed that namespaces are not nested and all namespaces are on the same level below root. So more an implicit assumption (that is not true anymore).

Returning the root namespace when it's missing in context sounds like we are just one little bug away from root access 🙂

Reply:
Ah, I see, we're in an OpenBao test file. I was thinking this was an upstream test file that were modifying to add a root context where there was none before, indicating to me that we had a behavior difference.

But I think we're fine. I concur though, we could be one bug away from root access, though I suppose that's true anyways since we'd always start at the root namespace anyways. :-)